hooks/useClickOutside.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useRef } from 'react';
+
+type useClickOutsideType = (
+  // eslint-disable-next-line no-unused-vars
+  event: MouseEvent | TouchEvent
+) => void;
+
+export const useClickOutside = <T extends HTMLElement>(
+  handler: useClickOutsideType
+) => {
+  const ref = useRef<T>();
+  const clickEvent = 'ontouchstart' in window ? 'touchstart' : 'click';
```

Comment:
Since `clickEvent` isn't used inside the render function, we should move it inside the `useEffect`